cpp/include/cudf/copying.hpp

```diff
@@ -331,19 +376,155 @@ std::vector<gdf_column*> slice(gdf_column const &          input_column,
+ * of the columns of input tables and the output tables must be the same.
+ *
+ * Example:
+ * input:   [{10, 12, 14, 16, 18, 20, 22, 24, 26, 28}, 
+ *           { 1,  2,  3,  4, null, 0, 2,  4,  6,  2}]
+ * splits: {2, 5, 9}
+ * output:  { [{10, 12}, {1, 2}], [{14, 16, 18}, {3, 4, null}], 
+ *            [{20, 22, 24, 26}, {0, 2, 4, 6}], [{28}, {2}]}
+ *
+ * @param[in] input_table  The input table whose rows will be split.
+ * @param[in] splits       An device array of indices that are used to divide
+ * @param[in] num_splits   Number of splits in the splits indices array
+ * the input table into multiple tables.
+ * @return A std::vector of cudf::table, each of which may have a different size
+ * a different number of rows.
+ */
+std::vector<cudf::table> split(cudf::table const &        input_table,
+                               gdf_index_type const*      splits,
+                               gdf_size_type              num_splits);
 
+/**
+ * @brief Scatters the rows of a table to `n` tables according to a scatter map
+ *
+ * This 'scatter_to_tables' function copies the rows from the input table to new
+ * tables according to the table indices given by scatter map.
+ * The number of output tables will be equal to n = (1 + maximum of scatter map
+ * values)
+ * If a value in [0,n] does not appear in scatter_map, then the corresponding
+ * output table will be empty.
+ *
+ * `scatter_map` is a non-nullable column of `GDF_INT32` elements whose `size`
+ * equals `input.num_rows()` and contains numbers in range of [0, n].
+ * The datatypes of the columns of input tables and the output tables must be
+ * the same.
+ *
+ * Exceptional cases for the scatter_map column are:
+ * @throws cudf::logic_error when `scatter_map.dtype != GDF_INT32`
+ * @throws cudf::logic_error when `scatter_map.size != input_table.num_rows()`
+ * @throws cudf::logic_error when `has_nulls(scatter_map) == true`
+ *
+ * Example:
+ * input:       [{10, 12, 14, 16, 18, 20, 22, 24, 26, 28}, 
+ *               { 1,  2,  3,  4, null, 0, 2,  4,  6,  2}]
+ * scatter_map:  { 3,  4,  3,  1,  4,  4,  0,  1,  1,  1}
+ * output:     {[{22}, {2}], 
+ *              [{16, 24, 26, 28}, {4, 4, 6, 2}], 
+ *              [{}, {}], 
+ *              [{10, 14}, {1, 3}], 
+ *              [{12, 18, 20}, {2, null, 0}]}
+ *
+ * @param[in] input_table  Table whose rows will be partitioned into a set of
+ * tables according to `scatter_map` 
+ * @param[in] scatter_map  Non-nullable column of `GDF_INT32` values that map
+ * each row in `input_table` into one of the output tables. 
+ *
+ * @return A std::vector of `table`s containing the scattered rows of `input_table`.
+ * `table` `i` contains all rows `j` from `input` where `scatter_map[j] == i`. 
+ *
+ */
+std::vector<cudf::table>
+scatter_to_tables(cudf::table const& input_table, gdf_column const& scatter_map);
+
+ /**
+ * @brief Groups the rows of a table into `n` tables according to a
```

Comment:
This isn't a `scatter` operation, it's a groupby. A scatter implies that if `scatter_map[i] == j`, then row `i` from `input_table` will be scattered to table `j`. 

The difference with `groups_to_tables` is that that the `scatter_map` should really be a `table` called `keys`.

Reply:
moved `groups_to_tables` out of this PR.